app_dart/test/service/luci_build_service_test.dart

```diff
@@ -485,140 +487,138 @@ void main() {
+    test('Rerun a test failed flutter builder', () async {
+      config.db.values[totCommit.key] = totCommit;
       config.maxLuciTaskRetriesValue = 1;
       const LuciTask luciTask = LuciTask(
           commitSha: 'abc',
           ref: 'refs/heads/master',
-          status: Task.statusInfraFailure,
+          status: Task.statusFailed,
           buildNumber: 1,
           builderName: 'Mac abc',
-          summaryMarkdown: 'Step(\'checkout source code.Checkout flutter/flutter.git fetch\') (retcode: -9)');
+          summaryMarkdown: 'summary');
       final bool rerunFlag = await service.checkRerunBuilder(
-        commitSha: 'abc',
+        commit: commit,
         luciTask: luciTask,
         retries: 0,
         repo: 'flutter',
+        datastore: datastore,
       );
       expect(rerunFlag, true);
     });
 
-    test('Rerun a flutter Mac builder with infra failure: -9 in shards', () async {
+    test('Rerun an infra failed flutter builder', () async {
+      config.db.values[totCommit.key] = totCommit;
       config.maxLuciTaskRetriesValue = 1;
       const LuciTask luciTask = LuciTask(
           commitSha: 'abc',
           ref: 'refs/heads/master',
           status: Task.statusInfraFailure,
           buildNumber: 1,
-          builderName: 'Mac tool_tests',
-          summaryMarkdown:
-              'recipe infra failure: Infra Failure: Step(\'display builds.build(s) failed\') (retcode: 1)');
+          builderName: 'Mac abc',
+          summaryMarkdown: 'summary');
       final bool rerunFlag = await service.checkRerunBuilder(
-        commitSha: 'abc',
+        commit: commit,
         luciTask: luciTask,
         retries: 0,
         repo: 'flutter',
+        datastore: datastore,
       );
       expect(rerunFlag, true);
     });
 
-    test('Do not rerun a flutter Mac builder with infra failure: -9 in shards but not supported builder', () async {
+    test('Do not rerun a successful flutter builder', () async {
+      config.db.values[totCommit.key] = totCommit;
       config.maxLuciTaskRetriesValue = 1;
       const LuciTask luciTask = LuciTask(
           commitSha: 'abc',
           ref: 'refs/heads/master',
-          status: Task.statusInfraFailure,
+          status: Task.statusSucceeded,
           buildNumber: 1,
-          builderName: 'Mac abc',
-          summaryMarkdown:
-              'recipe infra failure: Infra Failure: Step(\'display builds.build(s) failed\') (retcode: 1)');
+          builderName: 'Mac abc');
       final bool rerunFlag = await service.checkRerunBuilder(
-        commitSha: 'abc',
+        commit: commit,
         luciTask: luciTask,
         retries: 0,
         repo: 'flutter',
+        datastore: datastore,
       );
       expect(rerunFlag, false);
     });
 
-    test('Do not rerun a flutter Mac builder with non -9 retcode', () async {
+    test('Do not rerun a non-supported repo builder', () async {
```

Comment:
To which use case does this test applies?

Reply:
Since the logic is for framework only, removing the test to avoid confusion.